code-challenges.js

```diff
@@ -40,12 +98,75 @@ var testString1 = "albatross"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
 
+// Write a function that takes a string
+const lowerThenUpper = (string) => {
+
+    //Split the string into an array
+    //Map over the array using a ternary operator making letters at even indices lowercase and letters at odd indices uppercase
+    //Join the letters back into a string
+    //Return the new string
+    return string.split("").map((letter, index) => index % 2 === 0 ? letter.toLowerCase() : letter.toUpperCase()).join("")
+}
 
 
+console.log("#4:");
+console.log(lowerThenUpper(testString1));// expect: "aLbAtRoSs"
+console.log(lowerThenUpper(testString2), "\n");// expect: "jAbBeRwOcKy"
 
 
 // --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 var arr1 = [3, 7, 10, 5, 4, 3, 3]
 var arr2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+//Write a function that takes two arrays
+const noDupeNumbers = (array1, array2) => {
+    //Create an empty array to hold the answer
+    answerArray = []
+    //Create a function to loop through an array
+    const pushFirstInstance = (array) => {
+        //Iterate over the length of the array from first element to last
+        for (let i = 0; i < array.length; i++) {
+            //Push each element to the answer array if it is not already present
+            answerArray.includes(array[i]) ? "" : answerArray.push(array[i])
+        }
+    }
```

Comment:
Logic looks great. A good refactor would be a higher-order function like filter.